lib/node_modules/@stdlib/iter/while-each/README.md

```diff
@@ -0,0 +1,243 @@
+var array2iterator = require( '@stdlib/array/to-iterator' );
+
+function predicate( v ) {
+    return v < 3;
+}
+
+function assert( v ) {
+    if ( v !== v ) {
+        throw new Error( 'should not be NaN' );
+    }
+}
+
+var it = iterWhileEach( array2iterator( [ 1, 2, 3, 4 ] ), predicate, assert );
+// returns <Object>
+
+var r = it.next().value;
+// returns 1
+
+r = it.next().value;
+// returns 2
+
+r = it.next().value;
+// undefined
+
+// ...
+```
+
+To set the function execution context, provide a `thisArg`.
+
+```javascript
+var array2iterator = require( '@stdlib/array/to-iterator' );
+
+function assert( v ) {
+    this.count += 1;
+    if ( v !== v ) {
+        throw new Error( 'should not be NaN' );
+    }
+}
+
+function predicate( v ) {
+    return v < 3;
+}
+
+var ctx = {
+    'count': 0
+};
+
+var it = iterWhileEach( array2iterator( [ 1, 2, 3 ] ), predicate, assert, ctx );
+// returns <Object>
+
+var r = it.next().value;
+// returns 1
+
+r = it.next().value;
+// returns 2
+
+var count = ctx.count;
+// returns 2
+```
+
+</section>
+
+<!-- /.usage -->
+
+<!-- Package usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="notes">
+
+## Notes
+
+-   If an environment supports `Symbol.iterator` **and** a provided iterator is iterable, the returned iterator is iterable.
+
+</section>
+
+<!-- /.notes -->
+
+<!-- Package usage examples. -->
+
+<section class="examples">
+
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var randu = require( '@stdlib/random/iter/randu' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var iterForEach = require( '@stdlib/iter/for-each' );
+
+function assert( v ) {
+    if ( isnan( v ) ) {
+        throw new Error( 'should not be NaN' );
+    }
+}
+
+function predicate( v ) {
+    return v < 0.5;
+}
+
+// Create a seeded iterator for generating pseudorandom numbers:
+var rand = randu({
+    'seed': 1234,
+    'iter': 10
+});
+
+// Create an iterator which validates generated numbers:
+var it = iterForEach( rand, predicate, assert );
+
+// Perform manual iteration...
+var r;
+while ( true ) {
+    r = it.next();
+    if ( r.done ) {
+        break;
+    }
+    console.log( r.value );
+}
+```
+
+</section>
+
+<!-- /.examples -->
+
+<!-- Section to include cited references. If references are included, add a horizontal rule *before* the section. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="references">
+
+</section>
+
+<!-- /.references -->
+
+<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
+
+<section class="related">
+
+* * *
+
+## See Also
+
+-   <span class="package-name">[`@stdlib/iter/map`][@stdlib/iter/map]</span><span class="delimiter">: </span><span class="description">create an iterator which invokes a function for each iterated value.</span>
```

Comment:
```suggestion
```

We handle "See Also" sections in a separate build process which draws from a central project database.